packages/scm/src/browser/scm-tree-widget.tsx

```diff
@@ -63,50 +61,53 @@ export class ScmTreeWidget extends TreeWidget {
         @inject(TreeProps) readonly props: TreeProps,
         @inject(ScmTreeModel) readonly model: ScmTreeModel,
         @inject(ContextMenuRenderer) protected readonly contextMenuRenderer: ContextMenuRenderer,
-        @inject(ScmService) protected readonly scmService: ScmService,
     ) {
         super(props, model, contextMenuRenderer);
         this.id = ScmTreeWidget.ID;
         this.addClass('groups-outer-container');
     }
 
+    protected visibleResourceNodes: ScmFileChangeNode[];
```

Comment:
Is it necessary to cache nodes here? Could we compute nodes from the model instead?

The model should be the source of truth, widget should not keep the model state.